src/pages/posts/[postid]/edit.astro

```diff
@@ -0,0 +1,43 @@
+---
+import { EditPostForm } from '../../../components/post-form/post-form';
+import Seo from '../../../components/seo.astro';
+import BaseLayout from '../../../layouts/base-layout.astro';
+import { supabase } from '../../../util/supabase.astro';
+
+const err = (code: number) => {
+    Astro.response.status = code;
+    Astro.response.statusText =
+        { 400: 'Bad request', 404: 'Not found', 500: 'Internal server error' }[
+            code
+        ] ?? 'Error';
+    return new Response(null, Astro.response);
+};
+
+const editKey = Astro.url.searchParams.get('key');
+const { postid } = Astro.params;
+
+if (!postid) {
+    return err(404);
+}
+
+const { data: postInfo, error } = await supabase()
+    .from('posts')
+    .select()
+    .eq('id', postid)
+    .single();
+
+if (error || !editKey || editKey !== postInfo?.edit_key) {
+    return err(400);
```

Comment:
Should this be 403 Forbidden?

Reply:
Also - on my end, going to a page with no/wrong edit key just shows a white page. Some kind of proper 'you dont have permission to edit that page' kinda warning would be good.

Reply:
I think this looks to be a [vite bug}(https://github.com/withastro/astro/issues/4418#issuecomment-1228531339)? While in development mode, it looks like the server doesn't handle error pages correctly, though that might be out of date information. Either way, added custom 403, 404, and 500 pages.